src/ucp/api/ucp.h

```diff
@@ -1916,6 +1916,31 @@ ucs_status_t ucp_ep_create(ucp_worker_h worker, const ucp_ep_params_t *params,
 ucs_status_ptr_t ucp_ep_close_nb(ucp_ep_h ep, unsigned mode);
 
 
+/**
+ * @ingroup UCP_ENDPOINT
+ *
+ * @brief Non-blocking @ref ucp_ep_h "endpoint" closure.
+ *
+ * This routine releases the @ref ucp_ep_h "endpoint". The endpoint closure
+ * process depends on the selected @a mode.
+ *
+ * @param [in]  ep      Handle to the endpoint to close.
+ * @param [in]  param   Operation parameters, see @ref ucp_request_param_t
+ *
+ * @return NULL                 - The endpoint is closed successfully.
+ * @return UCS_PTR_IS_ERR(_ptr) - The closure failed and an error code indicates
+ *                                the transport level status. However, resources
+ *                                are released and the @a endpoint can no longer
+ *                                be used.
+ * @return otherwise            - The closure process is started, and can be
+ *                                completed at any point in time. A request
+ *                                handle is returned to the application in order
+ *                                to track progress of the endpoint closure.
+ */
+ucs_status_ptr_t ucp_ep_close_nbx(ucp_ep_h ep,
```

Comment:
pls document which flags could be specified in param->flags

Reply:
added docs

Reply:
@hoopoepg - Am I missing something, because I don't see where it was documented.

Reply:
@shamisp line 1927 of this file

Reply:
I think this is not what Yossi meant - aka specify what particular parameters are relevant for this operation.

Reply:
@yosefe any ideas?

Reply:
@shamisp I've meant what @hoopoepg added in line 1927 - this is the only particular parameter which is relevant for this operation